neurodsp/filt.py

```diff
@@ -67,204 +66,216 @@ def filter_signal(sig, fs, pass_type, fc, n_cycles=3, n_seconds=None,
-                cf_3db_2 = next(f_db[ind] for ind in range(len(db))[::-1] if db[ind] > -3)
-                # Compute transition bandwidth
-                transition_bw1 = cf_3db_1 - cf_20db_1
-                transition_bw2 = cf_20db_2 - cf_3db_2
-                transition_bw = max(transition_bw1, transition_bw2)
-
-            elif pass_type == 'bandstop':
-                pass_bw = f_hi - f_lo
-                # Identify edges of transition band (-3dB and -20dB)
-                cf_20db_1 = next(f_db[ind] for ind in range(len(db)) if db[ind] < -20)
-                cf_3db_1 = next(f_db[ind] for ind in range(len(db)) if db[ind] < -3)
-                cf_20db_2 = next(f_db[ind] for ind in range(len(db))[::-1] if db[ind] < -20)
-                cf_3db_2 = next(f_db[ind] for ind in range(len(db))[::-1] if db[ind] < -3)
-                # Compute transition bandwidth
-                transition_bw1 = cf_20db_1 - cf_3db_1
-                transition_bw2 = cf_3db_2 - cf_20db_2
-                transition_bw = max(transition_bw1, transition_bw2)
-
-            elif pass_type == 'highpass':
-                pass_bw = f_nyq - f_lo
-                # Identify edges of transition band (-3dB and -20dB)
-                cf_20db = next(f_db[ind] for ind in range(len(db)) if db[ind] > -20)
-                cf_3db = next(f_db[ind] for ind in range(len(db)) if db[ind] > -3)
-                # Compute transition bandwidth
-                transition_bw = cf_3db - cf_20db
-
-            elif pass_type == 'lowpass':
-                pass_bw = f_hi
-                # Identify edges of transition band (-3dB and -20dB)
-                cf_20db = next(f_db[ind] for ind in range(len(db)) if db[ind] < -20)
-                cf_3db = next(f_db[ind] for ind in range(len(db)) if db[ind] < -3)
-                # Compute transition bandwidth
-                transition_bw = cf_20db - cf_3db
-
-            print('Transition bandwidth is ' + str(np.round(transition_bw, 1)) + \
-                  ' Hz. Pass/stop bandwidth is ' + str(np.round(pass_bw, 1)) + ' Hz')
-
-            # Raise warning if transition bandwidth is too high
-            if transition_bw > pass_bw:
-                warnings.warn('Transition bandwidth is ' + str(np.round(transition_bw, 1)) + \
-                              ' Hz. This is greater than the desired pass/stop bandwidth of '\
-                               + str(np.round(pass_bw, 1)) + ' Hz')
+    # Compute the frequency response in terms of Hz and dB
+    w_vals, h_vals = signal.freqz(b_vals, a_vals)
+    f_db = w_vals * fs / (2. * np.pi)
+    db = 20 * np.log10(abs(h_vals))
 
-    # Remove edge artifacts
-    if not iir and remove_edge_artifacts:
-        n_rmv = int(np.ceil(filt_len / 2))
-        sig_filt[:n_rmv] = np.nan
-        sig_filt[-n_rmv:] = np.nan
+    # Confirm frequency response goes below -20dB (significant attenuation)
+    if np.min(db) >= -20:
```

Comment:
-20 and -3 are _kind of_ magic numbers, and we might want to expose them to users to define, and set defaults in undefined.

Reply:
fair. Yeah, I don't think there's any problem with making them extra kwargs.

Reply:
I added this as a set-able parameter in the sub-function that now checks filter properties, including transition bands. It's not currently added as an arg to all the top level filter functions, just because it's getting to be a lot of parameters. It could be added - or it could be a case of saying "call the transition function manually if you want to check custom transition bands"